src/utils/builder.ts

```diff
@@ -503,6 +604,53 @@ export function makeStateBranch(pathway: Pathway, stateKey: string): Pathway {
   };
 }
 
+export function createCQL(action: Action, stateKey: string): string {
```

Comment:
I'm a little confused by the content of this pull request... I know it's based on Keeyan's branch but the delta seems to be including a lot of UI items from that branch for some reason. For now though I'm assuming this function is the majority of what's relevant to look at here.

Reply:
I think he may have pushed again so I need to rebase again